### PR TITLE
Replace Git Subtree with Custom Git Hooks Exercise

### PR DESCRIPTION
**Pull request by mjaehn**
Solves https://github.com/C2SM/git-course/issues/56

### CODE REVIEW

File: advanced/Exercise_7.md [hunk 1]
```diff
@@ -1,154 +1,97 @@
-# Exercise 7 - Using git submodule
+# Exercise 7 - Custom Git Hooks
 
-This exercise is designed to be paired with Exercise 8 to help you compare and contrast the two main methods for embedding a git repository into another git repository. This exercise illustrates the submodule method, while the next exercise illustrates the subtree method. In this exercise, we will learn to use git submodule to nest a git repository inside another one. We will first add the git submodule and examine how the host and nested repositories deal with changes in their respective files. We will then learn how to pull and push changes to and from the sub-repository.
+In this exercise, you'll practice implementing a pre-commit hook in the `conference_planning` repository to prevent committing files with trailing whitespace. You'll also learn how to set up multiple pre-commit hooks for different tasks.
 
 This exercise uses the same git repository that was created in Exercise 3. If you have not already done so, you can create it by following the instructions in the ["Initialize the git repository" section of Exercise 3](./Exercise_3.md#initialize).
```
Comment (AnnikaLau) [inline advanced/Exercise_7.md:5]:
```suggestion
This exercise uses the same Git repository that was created in Exercise 3. If you have not already done so, you can create it by following the instructions in the [Initialize the Git repository](./Exercise_3.md#initialize) section of Exercise 3.
```

**Comment by AnnikaLau** [comment]
To be consistent with all other exercises, paths and branch names should be written in italic *like this* and commands `like this`

File: advanced/Exercise_7.md [hunk 2]
```diff
@@ -1,154 +1,97 @@
...
+## Implement Pre-Commit Hook for Trailing Whitespace <a name="whitespace"></a>
 
-Check the status of the repository.
+Create a new file named `pre-commit` in the `.git/hooks` directory. Make it executable.
 
-```plaintext
-git status
+```sh
+touch .git/hooks/pre-commit
+chmod +x .git/hooks/pre-commit
```
Comment (AnnikaLau) [inline advanced/Exercise_7.md:27]: Should we say something about `touch` and `chmod`? I'm sure there are people who don't know what that is.

Comment (mjaehn) [reply advanced/Exercise_7.md:27]: `touch` has been used in several exercises before. I will add a comment about `chmod +x".

File: advanced/Exercise_7.md [hunk 3]
```diff
@@ -1,154 +1,97 @@
...
-sed -i '/^11:00/a 12:00-13:00: Lunch break' schedule_day1
-```
-
-Check the status of the repository.
+# Redirect output to stderr.
+exec 1>&2
 
-```plaintext
-git status
+# Check for trailing whitespace in staged files.
```
Comment (AnnikaLau) [inline advanced/Exercise_7.md:40]: I don't know what "trailing whitespace" is, so I'm sure there are others who also don't know... A sentence about that would be good.

**Comment by mjaehn** [comment]
> To be consistent with all other exercises, paths and branch names should be written in italic _like this_ and commands `like this`

Who came up with this formatting guideline? I find inline code more readable for paths and branch names. Most commands are in code blocks anyway.

**Review by AnnikaLau — CHANGES_REQUESTED** [review]
just the few comments I had

File: advanced/Exercise_8.md [hunk 4]
```diff
@@ -1,59 +1,69 @@
-# Exercise 8 - Using git subtree
+# Exercise 8 - Using git submodule
 
-This exercise is designed to be paired with Exercise 7 to help you compare and contrast the two main methods for embedding a git repository into another git repository. In this exercise, we will learn to use `git subtree` to nest a git repository inside another one. We will first add the git subtree and examine how the host and nested repositories deal with changes in their respective files. We will then learn how to pull and push changes to and from the sub-repository.
+This exercise is designed to be paired with Exercise 8 to help you compare and contrast the two main methods for embedding a git repository into another git repository. This exercise illustrates the submodule method, while the next exercise illustrates the subtree method. In this exercise, we will learn to use git submodule to nest a git repository inside another one. We will first add the git submodule and examine how the host and nested repositories deal with changes in their respective files. We will then learn how to pull and push changes to and from the sub-repository.
```
Comment (AnnikaLau) [inline advanced/Exercise_8.md:3]: to be paired with Exercise 8?

File: advanced/Exercise_7.md (hunk 3 above)
Comment (mjaehn) [reply advanced/Exercise_7.md:42]: Solved in [9eb535f](https://github.com/C2SM/git-course/pull/83/commits/9eb535fb66e9256ff16b14ddb30ad04178322ca1).

File: advanced/Exercise_8.md (hunk 4 above)
Comment (mjaehn) [reply advanced/Exercise_8.md]: Removed in [2104b09](https://github.com/C2SM/git-course/pull/83/commits/2104b09d22dc5393175e7d12d63de02d12ea18f0)

**Comment by AnnikaLau** [comment]
Maybe we should also add an example to run a pre-commit without actually committing?

**Comment by mjaehn** [comment]
> Maybe we should also add an example to run a pre-commit without actually committing?

I don't think I understand your question.

**Comment by AnnikaLau** [comment]
> > Maybe we should also add an example to run a pre-commit without actually committing?
> 
> I don't think I understand your question.

Right now, the pre-commit only runs when one commits something. But one should also be able to run something like `pre-commit run --all-files --verbose` to see what the pre-commit would actually do.

**Comment by mjaehn** [comment]
> > > > > Maybe we should also add an example to run a pre-commit without actually committing?
> > > > 
> > > > 
> > > > I don't think I understand your question.
> > > 
> > > 
> > > Right now, the pre-commit only runs when one commits something. But one should also be able to run something like `pre-commit run --all-files --verbose` to see what the pre-commit would actually do.
> > 
> > 
> > Ok cool, do you want to implement that here or in a separate PR?
> 
> I would leave it to you to do it here :-P

Done 😸

File: advanced/Exercise_7.md [hunk 5]
```diff
@@ -1,154 +1,147 @@
...
+    entry: .git/hooks/pre-commit-whitespace
+    language: system
+  # Add other hooks here if needed
+```
 
-Navigate back to your fork of the poster repository on Github, and make a change to the schedule file and commit it. You can do this directly on the web interface by simply selecting the file and using the edit button.
+This configuration specifies that the `pre-commit-whitespace` hook (the one we created earlier) should run.
 
-Once you have done this, go to back to the terminal and use `git submodule update` to get the main repository to fetch the new commit of the poster repository.
+To manually run the `pre-commit` checks without making a commit, use the following command:
```
Comment (AnnikaLau) [inline advanced/Exercise_7.md:132]: It should be mentioned that this command only affects staged files.

Comment (mjaehn) [reply advanced/Exercise_7.md]: Done 👍🏻

**Review by AnnikaLau — APPROVED** [review]

**Pull request merged**